Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyAssetCtrl.cpp

```diff
@@ -770,7 +788,7 @@ namespace AzToolsFramework
                     if (!assetID.IsValid())
                     {
                         // No Asset Id selected - Open editor and create new asset for them
-                        AssetEditor::AssetEditorRequestsBus::Broadcast(&AssetEditor::AssetEditorRequests::CreateNewAsset, GetCurrentAssetType());
+                        AssetEditor::AssetEditorRequestsBus::Broadcast(&AssetEditor::AssetEditorRequests::CreateNewAsset, GetCurrentAssetType(), m_componentUuid);
```

Comment:
Perhaps I misunderstood the comment, but would it be possible to create the random UUID locally, connect to the asset editor notification bus here, and pass it in to the create new asset function, then disconnect when you receive the on created asset notification? Then you would not need to provide or store the UUID from the component.

const AZ::Uuid uuid = AZ::Uuid::CreateRandom();
AssetEditor::AssetEditorNotificationsBus::Handler::BusConnect(uuid);
AssetEditor::AssetEditorRequestsBus::Broadcast(&AssetEditor::AssetEditorRequests::CreateNewAsset, GetCurrentAssetType(), uuid);

Reply:
Unfortunately that won't work, as the PropertyAssetControl gets torn down and rebuilt when the PropertyEditor is invalidated by the AssetEditorWidget after the asset is created in memory, but before it is added to the catalog and the notification is received.